src/main/java/com/puppycrawl/tools/checkstyle/gui/CodeSelectorPresentation.java

```diff
@@ -45,22 +43,22 @@
      * Constructor.
      * @param ast ast node.
      * @param lines2position list to map lines.
+     * @noinspection AssignmentToCollectionOrArrayFieldFromParameter
      */
     public CodeSelectorPresentation(DetailAST ast, List<Integer> lines2position) {
         node = ast;
-        final List<Integer> copy = new ArrayList<>(lines2position);
-        this.lines2position = Collections.unmodifiableList(copy);
+        this.lines2position = lines2position;
     }
 
     /**
      * Constructor.
      * @param node DetailNode node.
      * @param lines2position list to map lines.
+     * @noinspection AssignmentToCollectionOrArrayFieldFromParameter
      */
     public CodeSelectorPresentation(DetailNode node, List<Integer> lines2position) {
```

Comment:
argument type should ImmutableList, in other case it is not clear why we suppress violation